### DISCUSSION TITLE
How to override Chromedriver

### DISCUSSION THREAD

**Discussion by drFrankWhite** [seleniumbase/SeleniumBase#2310] [Q&A] [answered]
I've been added to a QA team that has a test project that uses the selenium(arm)/standalone-chromium image to run their UI tests.

I would like to use SB, but I'm having trouble getting SB to run with their existing Chromedriver.

**Comment by mdmintz** [comment, marked as answer]
SeleniumBase automatically takes care of driver management. All you have to do is decide on the syntax format that you want to use from [SeleniumBase/help_docs/syntax_formats.md](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md).

For example, the `SB()` context manager format:

```python
from seleniumbase import SB

with SB() as sb:
    sb.open("seleniumbase.io/simple/login")
    sb.type("#username", "demo_user")
    sb.type("#password", "secret_pass")
    sb.click('a:contains("Sign in")')
    sb.assert_exact_text("Welcome!", "h1")
    sb.assert_element("img#image1")
    sb.highlight("#image1")
    sb.click_link("Sign out")
    sb.assert_text("signed out", "#top_message")
```

Here's a raw `driver` format:

```python
driver = Driver()
try:
    driver.open("seleniumbase.github.io/demo_page")
    driver.highlight("h2")
    driver.type("#myTextInput", "Automation")
    driver.click("#checkBox1")
    driver.highlight("img", loops=6)
finally:
    driver.quit()
```

Here's a `pytest` format:

```python
from seleniumbase import BaseCase
BaseCase.main(__name__, __file__)

class TestSimpleLogin(BaseCase):
    def test_simple_login(self):
        self.open("seleniumbase.io/simple/login")
        self.type("#username", "demo_user")
        self.type("#password", "secret_pass")
        self.click('a:contains("Sign in")')
        self.assert_exact_text("Welcome!", "h1")
        self.assert_element("img#image1")
        self.highlight("#image1")
        self.click_link("Sign out")
        self.assert_text("signed out", "#top_message")
```

There are lots of command-line options that you can use to change default settings: [SeleniumBase/help_docs/customizing_test_runs.md](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/customizing_test_runs.md)
(This lets you enable different modes, such as `--headless` mode, `--demo` mode, etc.)

**Reply by drFrankWhite** [reply]
Maybe my question should have been: Can I get SB's Chromedriver to be the Docker selenium/standalone-chrome:latest image I currently have connected to GitLab jobs, and which is used for a pile of legacy tests that I don't want to re-write?

**Reply by mdmintz** [reply]
Your wording is a bit confusing because your comparing a driver (`chromedriver`) with a Docker image (`selenium/standalone-chrome:latest`). If you want to get `seleniumbase` in your Docker container, just run `pip install seleniumbase` from your Docker container, and then run a SeleniumBase script, which will automatically download a `chromedriver` that matches your version of Chrome if there isn't already one on the PATH.

**Reply by drFrankWhite** [reply]
My apologies, I know I'm all over the place.

The legacy tests use regular old selenium to write their UI tests, and they're using the selenium/standalone-chrome:latest image to run them.  But when I try to run my sb tests with the same `--proxy=123.45.6.7:4444` flag, I bump into "ERR_PROXY_CONNECTION_FAILED"

I guess I'm trying to run my new SB tests with the same image, but I'm starting to think it isn't gonna work.

**Reply by mdmintz** [reply]
If you're using one the of `pytest` formats, then your `pytest --proxy=123.45.6.7:4444` is a valid command-line option for that. I'm not sure which format you're using, so I'm not sure what info you're specifically looking for yet. The formats can be found on [SeleniumBase/help_docs/syntax_formats.md](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md). If you're not using one of the `pytest` formats, then proxy is set via method arg. Eg:

```python
from seleniumbase import Driver

driver = Driver(proxy="123.45.6.7:4444")
```
